logrotate.c

```diff
@@ -839,6 +840,9 @@ static int compressLogFile(const char *name, const struct logInfo *log, const st
 
     if (setSecCtx(inFile, name, &prevCtx) != 0) {
         /* error msg already printed */
+#ifdef WITH_SELINUX
+        freecon(prevCtx);
```

Comment:
Maybe move the cleanup into `setSecCtx()`?

Reply:
Thank you, that's a good point.

Please check if my patches don't do more harm than good. I am still new to the logrotate code.